Modelica/Math/isPowerOf2.mo

```diff
@@ -3,9 +3,17 @@ function isPowerOf2 "Determine if the integer input is a power of 2"
   extends Modelica.Icons.Function;
   input Integer i(min=1) "Integer scalar";
   output Boolean result "= true, if integer scalar is a power of 2";
+protected 
+  Integer power;
+  Integer halfPower;
 algorithm
   assert(i>=1, "Integer input to isPowerOf2 has to be >= 1");
-  result := i == 2^integer(log(i)/log(2)+0.5);
+  power:=integer(log(i)/log(2)+0.5);
+  halfPower:=integer(0.5+2^div(power,2));
```

Comment:
This has the same problem as the other PR:  As long as one can't trust `pow` in Modelica, one has to avoid using it when the result matters.  It's clear to me that this needs a discussion on its own, so I'll open a specification PR for that.

The key step here is still to compute an `Integer` value of a (potentially large) power of 2, and to me the added complexity with the intermediate power `div(power, 2)` doesn't seem to add any value over the first and simpler PR.